// @ankr.com/stkr-jssdk/dist/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoteStatus = exports.BlockchainNetworkId = exports.ProposalStatus = void 0;
var ProposalStatus;
(function (ProposalStatus) {
    ProposalStatus[ProposalStatus["Voting"] = 0] = "Voting";
    ProposalStatus[ProposalStatus["Fail"] = 1] = "Fail";
    ProposalStatus[ProposalStatus["Passed"] = 2] = "Passed";
})(ProposalStatus = exports.ProposalStatus || (exports.ProposalStatus = {}));
(function (ProposalStatus) {
    ProposalStatus[ProposalStatus["VOTING"] = 1] = "VOTING";
    ProposalStatus[ProposalStatus["FAIL"] = 2] = "FAIL";
    ProposalStatus[ProposalStatus["PASS"] = 3] = "PASS";
})(ProposalStatus = exports.ProposalStatus || (exports.ProposalStatus = {}));
var BlockchainNetworkId;
(function (BlockchainNetworkId) {
    BlockchainNetworkId[BlockchainNetworkId["mainnet"] = 1] = "mainnet";
    BlockchainNetworkId[BlockchainNetworkId["ropsten"] = 3] = "ropsten";
    BlockchainNetworkId[BlockchainNetworkId["rinkeby"] = 4] = "rinkeby";
    BlockchainNetworkId[BlockchainNetworkId["goerli"] = 5] = "goerli";
    BlockchainNetworkId[BlockchainNetworkId["dev"] = 2018] = "dev";
    BlockchainNetworkId[BlockchainNetworkId["classic"] = 61] = "classic";
    BlockchainNetworkId[BlockchainNetworkId["mordor"] = 63] = "mordor";
    BlockchainNetworkId[BlockchainNetworkId["kotti"] = 6] = "kotti";
})(BlockchainNetworkId = exports.BlockchainNetworkId || (exports.BlockchainNetworkId = {}));
var VoteStatus;
(function (VoteStatus) {
    VoteStatus["NO"] = "0x564f54455f4e4f";
    VoteStatus["YES"] = "0x564f54455f594553"; // web3.utils.fromAscii("VOTE_YES")
})(VoteStatus = exports.VoteStatus || (exports.VoteStatus = {}));
