// @ankr.com/stkr-jssdk/dist/contracts/AnkrDeposit.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnkrDepositEvents = exports.AnkrDeposit = void 0;
var AnkrDeposit_json_1 = __importDefault(require("./abi/AnkrDeposit.json"));
var BaseContract_1 = require("./BaseContract");
var EventBase_1 = require("../EventBase");
var AnkrDeposit = /** @class */ (function (_super) {
    __extends(AnkrDeposit, _super);
    function AnkrDeposit(web3, network) {
        var _this = _super.call(this, web3, network) || this;
        _this.abi = AnkrDeposit_json_1.default;
        _this.events = new AnkrDepositEvents(_this.getContract());
        return _this;
    }
    AnkrDeposit.prototype.getName = function () {
        return 'AnkrDeposit';
    };
    AnkrDeposit.prototype.owner = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.getContract().methods['owner']().call()];
            });
        });
    };
    AnkrDeposit.prototype.renounceOwnership = function (options) {
        return this.getContract().methods['renounceOwnership']().send(options);
    };
    AnkrDeposit.prototype.transferOwnership = function (newOwner, options) {
        return this.getContract().methods['transferOwnership'](newOwner).send(options);
    };
    AnkrDeposit.prototype.deposit = function (options) {
        return this.getContract().methods['deposit']().send(options);
    };
    AnkrDeposit.prototype.deposit_1 = function (user, options) {
        return this.getContract().methods['deposit_1'](user).send(options);
    };
    AnkrDeposit.prototype.withdraw = function (amount, options) {
        return this.getContract().methods['withdraw'](amount).send(options);
    };
    AnkrDeposit.prototype.unfreeze = function (addr, amount, options) {
        return this.getContract().methods['unfreeze'](addr, amount).send(options);
    };
    AnkrDeposit.prototype.freeze = function (addr, amount, options) {
        return this.getContract().methods['freeze'](addr, amount).send(options);
    };
    AnkrDeposit.prototype.availableDepositsOf = function (user, options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.getContract().methods['availableDepositsOf'](user).call()];
            });
        });
    };
    AnkrDeposit.prototype.depositsOf = function (user, options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.getContract().methods['depositsOf'](user).call()];
            });
        });
    };
    AnkrDeposit.prototype.frozenDepositsOf = function (user, options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.getContract().methods['frozenDepositsOf'](user).call()];
            });
        });
    };
    AnkrDeposit.prototype.updateGovernance = function (governance, options) {
        return this.getContract().methods['updateGovernance'](governance).send(options);
    };
    AnkrDeposit.prototype.allowAddressForFunction = function (addr, topic, options) {
        return this.getContract().methods['allowAddressForFunction'](addr, topic).send(options);
    };
    return AnkrDeposit;
}(BaseContract_1.BaseContract));
exports.AnkrDeposit = AnkrDeposit;
var AnkrDepositEvents = /** @class */ (function (_super) {
    __extends(AnkrDepositEvents, _super);
    function AnkrDepositEvents() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AnkrDepositEvents.prototype.getPastCompensate = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                // @ts-ignore
                return [2 /*return*/, this.contract.getPastEvents('Compensate', options)];
            });
        });
    };
    AnkrDepositEvents.prototype.getPastDeposit = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                // @ts-ignore
                return [2 /*return*/, this.contract.getPastEvents('Deposit', options)];
            });
        });
    };
    AnkrDepositEvents.prototype.getPastFreeze = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                // @ts-ignore
                return [2 /*return*/, this.contract.getPastEvents('Freeze', options)];
            });
        });
    };
    AnkrDepositEvents.prototype.getPastOwnershipTransferred = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                // @ts-ignore
                return [2 /*return*/, this.contract.getPastEvents('OwnershipTransferred', options)];
            });
        });
    };
    AnkrDepositEvents.prototype.getPastUnfreeze = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                // @ts-ignore
                return [2 /*return*/, this.contract.getPastEvents('Unfreeze', options)];
            });
        });
    };
    AnkrDepositEvents.prototype.getPastWithdraw = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                // @ts-ignore
                return [2 /*return*/, this.contract.getPastEvents('Withdraw', options)];
            });
        });
    };
    AnkrDepositEvents.prototype.Compensate = function (options) {
        return this.contract.events.Compensate(options);
    };
    AnkrDepositEvents.prototype.Deposit = function (options) {
        return this.contract.events.Deposit(options);
    };
    AnkrDepositEvents.prototype.Freeze = function (options) {
        return this.contract.events.Freeze(options);
    };
    AnkrDepositEvents.prototype.OwnershipTransferred = function (options) {
        return this.contract.events.OwnershipTransferred(options);
    };
    AnkrDepositEvents.prototype.Unfreeze = function (options) {
        return this.contract.events.Unfreeze(options);
    };
    AnkrDepositEvents.prototype.Withdraw = function (options) {
        return this.contract.events.Withdraw(options);
    };
    return AnkrDepositEvents;
}(EventBase_1.EventBase));
exports.AnkrDepositEvents = AnkrDepositEvents;
